### DISCUSSION TITLE
Use as NPM library

### DISCUSSION THREAD

**Discussion by Flavien** [mfucci/node-matter#126] [Q&A]
Is this package meant to be used as a library (as opposed to a command line tool)?

Right now I have to do:

    import { MatterDevice } from "node-matter/build/matter/MatterDevice";

It would be simpler if I could do:

    import { MatterDevice } from "node-matter";

Is there a way to package it this way for NPM?

**Comment by mfucci** [comment]
Yes, it is meant to be used as a library but I didn't configure it correctly so right now it is not really possible...
An index.js is missing exporting all the public APIs.

To add it, the problem is that the public APIs are not defined yet.

As you can see in Device.ts, my goal was to make MatterDevice the public object, but to use it tons of other classes need to be referenced...
I can right an experimental index.ts that exposes nearly everything, but next versions will change what is available.

If you have a suggestion for the library Apis, let me know!

**Comment by mfucci** [comment]
Hi Flavien,

I create a PR to add MatterDevice in an index.ts, so can you use:
import { MatterDevice } from "node-matter";

https://github.com/mfucci/node-matter/pull/131

You can use it before I commit it to make sure that it is working for your needs:

1. clone node-matter
2. checkout the branch "add-index-ts"
3. create a symlink inside the node-modules of your project pointing to the root of node-matter repository

If things don't work, modify index.ts or node-matter package.json and send me a PR with the modifications.

Sorry, I have little time right now to support further on this.

However, keep in mind the node-matter public APIs are far from being stable, so you might to update your code for new versions...
